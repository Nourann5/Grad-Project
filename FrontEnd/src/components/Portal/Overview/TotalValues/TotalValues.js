import React from 'react'
import styles from './TotalValues.module.css'
import {ReactComponent as CoursePlanPortalIcon} from 'assets/icons/coursePlanPortal.svg'
import {ReactComponent as ExamsPortalIcon} from 'assets/icons/examsPortal.svg'
import {ReactComponent as GroupsPortalIcon} from 'assets/icons/groupsPortal.svg'
import {ReactComponent as MembersPortalIcon} from 'assets/icons/membersPortal.svg'
function TotalValues() {
    return (
        <div className={styles['overview__total-values-statistics-wrapper']}>
            <div className={styles['overview__total-values-statistics']}>
                <div className={styles['overview__total-values-wrapper']}>
                    <h3 className={styles['overview__total-values-statistics-title']}>Total Tests</h3>
                    <h4 className={styles['overview__total-values-statistics-value']}>79</h4>
                </div>
                <div className={styles['overview__total-icon-wrapper']}>
                    <ExamsPortalIcon className={styles['overview__total-icon']}/>
                </div>
            </div>
            <div className={styles['overview__total-values-statistics']}>
                <div className={styles['overview__total-values-wrapper']}>
                    <h3 className={styles['overview__total-values-statistics-title']}>Total Courses</h3>
                    <h4 className={styles['overview__total-values-statistics-value']}>50</h4>
                </div>
                <div className={styles['overview__total-icon-wrapper']}>
                    <CoursePlanPortalIcon className={styles['overview__total-icon']}/>
                </div>
            </div>
            <div className={styles['overview__total-values-statistics']}>
                <div className={styles['overview__total-values-wrapper']}>
                    <h3 className={styles['overview__total-values-statistics-title']}>Total Groups</h3>
                    <h4 className={styles['overview__total-values-statistics-value']}>20</h4>
                </div>
                <div className={styles['overview__total-icon-wrapper']}>
                    <GroupsPortalIcon className={styles['overview__total-icon']}/>
                </div>
            </div>
            <div className={styles['overview__total-values-statistics']}>
                <div className={styles['overview__total-values-wrapper']}>
                    <h3 className={styles['overview__total-values-statistics-title']}>Total Students</h3>
                    <h4 className={styles['overview__total-values-statistics-value']}>1500</h4>
                </div>
                <div className={styles['overview__total-icon-wrapper']}>
                    <MembersPortalIcon className={styles['overview__total-icon']}/>
                </div>
            </div>
            
        </div>
    )
}

export default TotalValues