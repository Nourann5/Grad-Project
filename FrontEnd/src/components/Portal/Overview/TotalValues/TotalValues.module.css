.overview__total-values-statistics-wrapper{
    display: grid;
    grid-template-columns: repeat(4,1fr);
    grid-gap: 20px;
    margin-bottom: 30px;
}
.overview__total-values-statistics{
    background-color: white;
    border-radius: 15px;
    display: flex;
    align-items: center;
    padding: 15px 20px;
    transition: var(--transition);
    box-shadow: 0 0px 15px 1px rgb(0 0 0 / 10%);
}
.overview__total-values-statistics:hover{
    box-shadow: none;
    transform: translateY(-3px);
}

.overview__total-values-statistics-title{
    font-size: 12px;
    margin: 0;
    margin-bottom: 3px;
    color: #a09595;
}
.overview__total-values-statistics-value{
    font-weight: bold;
    font-size: 23px;
    color: var(--main-color);
}
.overview__total-icon-wrapper{
    align-items: center;
    justify-content: center;
    background: var(--main-color);
    color: white;
    padding: 10px;
    display: flex;
    border-radius: 15px;
    margin-left: auto;
}
.overview__total-icon path{
    fill:white
}

@media (max-width:992px){
    .overview__total-values-statistics-wrapper{
        grid-template-columns: repeat(2,1fr);
        grid-gap: 10px;
    }
}
@media (max-width: 426px){
    .overview__total-values-statistics-wrapper{
        grid-template-columns: repeat(1,1fr);
        grid-gap: 10px;
    }
}