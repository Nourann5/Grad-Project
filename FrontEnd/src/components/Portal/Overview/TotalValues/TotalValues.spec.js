import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import TotalValues from './TotalValues'

describe('Test the Total Value', () => { 
    it('heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <TotalValues/>
            </BrowserRouter>
        )
        expect(tree).toMatchSnapshot()
    })
 })