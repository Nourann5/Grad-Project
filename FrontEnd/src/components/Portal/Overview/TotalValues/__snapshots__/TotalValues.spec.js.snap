// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Total Value heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overview__total-values-statistics-wrapper"
      >
        <div
          class="overview__total-values-statistics"
        >
          <div
            class="overview__total-values-wrapper"
          >
            <h3
              class="overview__total-values-statistics-title"
            >
              Total Tests
            </h3>
            <h4
              class="overview__total-values-statistics-value"
            >
              79
            </h4>
          </div>
          <div
            class="overview__total-icon-wrapper"
          >
            <svg
              class="overview__total-icon"
            >
              examsPortal.svg
            </svg>
          </div>
        </div>
        <div
          class="overview__total-values-statistics"
        >
          <div
            class="overview__total-values-wrapper"
          >
            <h3
              class="overview__total-values-statistics-title"
            >
              Total Courses
            </h3>
            <h4
              class="overview__total-values-statistics-value"
            >
              50
            </h4>
          </div>
          <div
            class="overview__total-icon-wrapper"
          >
            <svg
              class="overview__total-icon"
            >
              coursePlanPortal.svg
            </svg>
          </div>
        </div>
        <div
          class="overview__total-values-statistics"
        >
          <div
            class="overview__total-values-wrapper"
          >
            <h3
              class="overview__total-values-statistics-title"
            >
              Total Groups
            </h3>
            <h4
              class="overview__total-values-statistics-value"
            >
              20
            </h4>
          </div>
          <div
            class="overview__total-icon-wrapper"
          >
            <svg
              class="overview__total-icon"
            >
              groupsPortal.svg
            </svg>
          </div>
        </div>
        <div
          class="overview__total-values-statistics"
        >
          <div
            class="overview__total-values-wrapper"
          >
            <h3
              class="overview__total-values-statistics-title"
            >
              Total Students
            </h3>
            <h4
              class="overview__total-values-statistics-value"
            >
              1500
            </h4>
          </div>
          <div
            class="overview__total-icon-wrapper"
          >
            <svg
              class="overview__total-icon"
            >
              membersPortal.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overview__total-values-statistics-wrapper"
    >
      <div
        class="overview__total-values-statistics"
      >
        <div
          class="overview__total-values-wrapper"
        >
          <h3
            class="overview__total-values-statistics-title"
          >
            Total Tests
          </h3>
          <h4
            class="overview__total-values-statistics-value"
          >
            79
          </h4>
        </div>
        <div
          class="overview__total-icon-wrapper"
        >
          <svg
            class="overview__total-icon"
          >
            examsPortal.svg
          </svg>
        </div>
      </div>
      <div
        class="overview__total-values-statistics"
      >
        <div
          class="overview__total-values-wrapper"
        >
          <h3
            class="overview__total-values-statistics-title"
          >
            Total Courses
          </h3>
          <h4
            class="overview__total-values-statistics-value"
          >
            50
          </h4>
        </div>
        <div
          class="overview__total-icon-wrapper"
        >
          <svg
            class="overview__total-icon"
          >
            coursePlanPortal.svg
          </svg>
        </div>
      </div>
      <div
        class="overview__total-values-statistics"
      >
        <div
          class="overview__total-values-wrapper"
        >
          <h3
            class="overview__total-values-statistics-title"
          >
            Total Groups
          </h3>
          <h4
            class="overview__total-values-statistics-value"
          >
            20
          </h4>
        </div>
        <div
          class="overview__total-icon-wrapper"
        >
          <svg
            class="overview__total-icon"
          >
            groupsPortal.svg
          </svg>
        </div>
      </div>
      <div
        class="overview__total-values-statistics"
      >
        <div
          class="overview__total-values-wrapper"
        >
          <h3
            class="overview__total-values-statistics-title"
          >
            Total Students
          </h3>
          <h4
            class="overview__total-values-statistics-value"
          >
            1500
          </h4>
        </div>
        <div
          class="overview__total-icon-wrapper"
        >
          <svg
            class="overview__total-icon"
          >
            membersPortal.svg
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
